### PR TITLE
pull creds into environment

### PR DESCRIPTION
**Pull request by slimslenderslacks**

### CODE REVIEW

**Comment by slimslenderslacks** [comment]
@cdupuis @ddgenome this is a change to extract artifactory creds into our leiningen builds.  I have a few improvements in mind (could be more generic so that we can populate any set of environment variables) but I'd like to get going with this and make sure that a few leiningen services are building correctly.

File: src/machine/leinSupport.ts [hunk 1]
```diff
@@ -75,13 +77,15 @@ export const LeinSupport: ExtensionPack = {
     },
 };
 
+const key = "(12 15 6 4 13 3 9 10 0 8 8 14 7 16 0 3)";
```
Comment (cdupuis) [inline src/machine/leinSupport.ts]: What's that? should that rather be a config value?

File: src/machine/leinSupport.ts [hunk 2]
```diff
@@ -92,13 +96,19 @@ function leinBuilder(projectLoader: ProjectLoader): Builder {
...
+                    return enriched;
+                },
                 projectToAppInfo: async (p: GitProject) => {
                     const projectClj = await p.findFile("project.clj");
                     logger.info(`run projectToAppInfo in ${p.baseDir}/${projectClj.path}`);
                     return {
                         name: clj.getName(`${p.baseDir}/${projectClj.path}`),
                         version: clj.getVersion(`${p.baseDir}/${projectClj.path}`),
-                        id: new GitHubRepoRef("owner", "repo"),
+                        id: new GitHubRepoRef( "owner", "repo"),
```
Comment (cdupuis) [inline src/machine/leinSupport.ts]: :-)

Comment (slimslenderslacks) [reply src/machine/leinSupport.ts]: ya, this is a todo but I can continue testing with this :).

File: src/machine/leinSupport.ts (hunk 1 above)
Comment (slimslenderslacks) [reply src/machine/leinSupport.ts]: ya, once this is working, I'll alter the lifecycle of this key.  But again, I just want to uncover the use cases here.

File: src/machine/leinSupport.ts [hunk 3]
```diff
@@ -35,7 +35,9 @@ import {
     asSpawnCommand,
     spawnAndWatch,
 } from "@atomist/sdm/util/misc/spawned";
+import { SpawnOptions } from "child_process";
 import * as df from "dateformat";
+import * as _ from "lodash/";
```
Comment (ddgenome) [inline src/machine/leinSupport.ts]: No need for a trailing slash there.

File: src/machine/leinSupport.ts [hunk 4]
```diff
@@ -75,13 +77,15 @@ export const LeinSupport: ExtensionPack = {
...
+const key = "(12 15 6 4 13 3 9 10 0 8 8 14 7 16 0 3)";
+
 function leinBuilder(projectLoader: ProjectLoader): Builder {
     return new SpawnBuilder(
         {
             projectLoader,
             options: {
                 name: "atomist.sh",
-                commands: [asSpawnCommand("./atomist.sh", {})],
+                commands: [asSpawnCommand("./atomist.sh", {env: {}})],
```
Comment (ddgenome) [inline src/machine/leinSupport.ts]: Does this work? Surely you need at least `HOME` set.

File: src/machine/leinSupport.ts [hunk 5]
```diff
@@ -92,13 +96,19 @@ function leinBuilder(projectLoader: ProjectLoader): Builder {
                         message: "lein errors",
                     };
                 },
+                enrich: async (options: SpawnOptions, p: GitProject): Promise<SpawnOptions> => {
+                    const encryptedEnv = {env: {ARTIFACTORY_USER: clj.vault(key, `${p.baseDir}/vault.txt`)["artifactory-user"],
+                                                ARTIFACTORY_PWD: clj.vault(key, `${p.baseDir}/vault.txt`)["artifactory-pwd"]}};
```
Comment (ddgenome) [inline src/machine/leinSupport.ts]: I have similar questions as @cdupuis on this.

File: src/machine/leinSupport.ts (hunk 3 above)
Comment (slimslenderslacks) [reply src/machine/leinSupport.ts]: okay, will remove that

File: src/machine/leinSupport.ts (hunk 4 above)
Comment (slimslenderslacks) [reply src/machine/leinSupport.ts]: the Process.env of the SDM is also merged somewhere else.  I'm not fully sure where the default environment responsibility sits at present.

File: src/machine/leinSupport.ts (hunk 5 above)
Comment (slimslenderslacks) [reply src/machine/leinSupport.ts]: I have questions about this as well to be honest.  I'm looking to find the highways here - this is a bit of a process.

**Pull request closed without merging**